### PR TITLE
Update limitations of event generator tool

### PR DESCRIPTION
**Pull request by santipadilla**
# Description

Related [issue](https://github.com/wazuh/wazuh-qa-automation/issues/83).

This PR is to add information about the limitations of the event generator tool. 

Includes: Event volume limitations, maximum generation rate, number of parallel event threads, other potential limitations and the results of testing performed.

### CODE REVIEW

**Review by rafabailon — APPROVED** [review]
LGTM

File: tests/performance/event_generator/src/event_generator/README.md [hunk 1]
```diff
@@ -20,3 +20,91 @@ The library is used by creating instances of the provided generator classes and
...
+| 100,000 | 10,000 | 10 | 10.035 | +0.035 |
+| 200,000 | 20,000 | 10 | 10.057 | +0.057 |
+| 250,000 | 25,000 | 10 | 10.482 | +0.482 |
+| 300,000 | 30,000 | 10 | 12.718 | +2.718 |
+| 500,000 | 50,000 | 10 | 16.229 | +6.229 |
+| 1,000,000 | 100,000 | 10 | 33.802 | +23.802 |
+
+- Below 250,000 operations, the observed execution times closely match the expected times.
+- Over 300,000 operations, the observed times start deviating from the expected times.
+- The deviation increases as both the operations and the rate increase, indicating system limitations.
```
Comment (Rebits) [inline tests/performance/event_generator/src/event_generator/README.md]: What operating systems and hardware have been tested so far? We need to determine the minimum requirements to ensure the tool performs effectively in real-world scenarios.

Additionally, the table could be updated to reflect EPS (Events Per Second) based on different host resource configurations, providing clear information about potential limitations.

File: tests/performance/event_generator/src/event_generator/README.md [hunk 2]
```diff
@@ -20,3 +20,91 @@ The library is used by creating instances of the provided generator classes and
...
+
+### CPU limitations
+
+- The generation of events involves processing that consumes CPU resources.
+- High rates and multiple threads can lead to high CPU utilization, impacting other system operations.
+
+### Memory usage
+
+- Each thread and its associated data structures consume memory.
+- Prolonged operation with many threads may lead to increased memory usage.
```
Comment (Rebits) [inline tests/performance/event_generator/src/event_generator/README.md]: Here’s an improved version of the message:

This documentation is too generic and does not adequately address the specific requirements for tool usage. To improve it, we need precise documentation that highlights real hardware limitations in practical environments. It should include actual data on the maximum EPS (Events Per Second) achieved under various resource configurations. This information will serve as a reference to determine whether the EPS usage in a given environment is feasible.

**Review by Rebits — CHANGES_REQUESTED** [review]
Good job, but some changes are required.

File: tests/performance/event_generator/src/event_generator/README.md (hunk 1 above)
Comment (santipadilla) [reply tests/performance/event_generator/src/event_generator/README.md]: Fixed [here](https://github.com/wazuh/wazuh-qa/pull/5842/files).

File: tests/performance/event_generator/src/event_generator/README.md (hunk 2 above)
Comment (santipadilla) [reply tests/performance/event_generator/src/event_generator/README.md]: Fixed [here](https://github.com/wazuh/wazuh-qa/pull/5842/files).

**Pull request closed without merging**